--- a/.iex.exs
+++ b/.iex.exs
@@ -2,3 +2,4 @@ alias HeadsUp.Repo
 alias HeadsUp.Incidents
 alias HeadsUp.Incidents.Incident
 alias HeadsUp.Admin
+import Ecto.Changeset
--- a/lib/heads_up/incidents/incident.ex
+++ b/lib/heads_up/incidents/incident.ex
@@ -17,5 +17,7 @@ defmodule HeadsUp.Incidents.Incident do
     incident
     |> cast(attrs, [:name, :description, :priority, :status, :image_path])
     |> validate_required([:name, :description, :priority, :status, :image_path])
+    |> validate_length(:description, min: 10)
+    |> validate_inclusion(:priority, 1..3)
   end
 end
